js/serial.js

```diff
@@ -12,10 +15,76 @@ var serial = {
     transmitting:   false,
     outputBuffer:  [],
 
+    LOGHEAD: 'SERIAL: ',
+
     connect: function (path, options, callback) {
         var self = this;
         self.openRequested = true;
 
+        var testUrl = path.match(/tcp:\/\/(.*):(.*)/)
```

Comment:
Be more specific here: `/^http:\/\/([A-Za-z0-9\.-]+)\:(\d+)$/`. This will also ensure that the port number is numeric.